src/compiler/crystal/syntax/parser.cr

```diff
@@ -3984,6 +3988,10 @@ module Crystal
             raise "cannot use '#{invalid_internal_name}' as a parameter name", invalid_internal_name
           end
           arg_name = external_name
+          if external_name_token.nil?
+            raise "missing external name token"
+          end
```

Comment:
When `external_name` is present `external_name_token` should also be present, so it may be better to turn this into an invariant.